Added Completed Task Strikethrough & Profile Popup via button in top left

CameronBill opened the pull request:
Popup currently contains form for changing a user's password

CameronBill committed:
Added Conditional Striking to task description on completion

CameronBill committed:
Added Completion changes from frontend repo

CameronBill committed:
Merge branch 'main' into cam-edits

CameronBill committed:
Added "Profile" Image

to be clicked on to show Change Password popup

CameronBill committed:
merged with main

CameronBill committed:
Made ChangePassword based on button not image

CameronBill committed:
Got Popup Working, Still an unexpected 404 when changing pword

CameronBill committed:
Got rid of the errors, but password still not changing

CameronBill committed:
Change Password works!

CameronBill committed:
Few Last minute changes

jubenjam commented on react-frontend/src/ProfilePopup.js:
```diff
@@ -0,0 +1,82 @@
+          <form>
+            <label htmlFor="oldPassword">Old Password</label>
+            <input
+              type="text"
```
Can you make the input type "password" instead of "text" (Similar to what was done in signup and login)

CameronBill replied:
Thank you, I was looking for what would cause that effect

jubenjam reviewed (changes requested):
Looks like it works. Just need to make the input type "password".

CameronBill committed:
Cleaned up backend, made change password text hidden

Removed redundant patch handler in backend, made form entry for profile popup type: password. Thanks Juan o/

CameronBill committed:
Idk why this folder is here

removed .vscode/ with empty settings.json file

jubenjam commented on react-frontend/src/ProfilePopup.js:
```diff
@@ -0,0 +1,82 @@
+      </Modal.Body>
+      <Modal.Footer>
+        <input
+          type="button"
```
instead of <input type="button"... 
you may want to use <button type="submit"...
This was used in the signup/login, and I think that that might be what is needed for just pressing enter to submit instead of clicking the button.

jubenjam replied:
Or nevermind, cause apparently that doesn't make that effect. no idea how Dustin did it then

jubenjam reviewed (commented):
Thanks for fixing that, looks good!

arigoodmann22 reviewed (approved):
good

yuneihlaing reviewed (approved):
Looks good!

dustint121 reviewed (approved):
Looks good. Feel free to merge.
